test/distributed/_sharded_tensor/ops/test_linear.py

```diff
@@ -90,23 +101,11 @@ def _run_sharded_linear(self, spec, input_size, linear_size, sharded_dim, merge_
             local_linear.weight, sharded_dim, TEST_GPU_NUM, spec, self.rank
         )
         local_grad_narrowed = local_grad.narrow(sharded_dim, start_pos, chunk_size)
-        local_weight_narrowed = local_linear.weight.narrow(
-            sharded_dim, start_pos, chunk_size
-        )
         local_bias_grad = local_linear.bias.grad
-        sharded_bias_grad = sharded_linear.bias.grad
-        # For col-wise sharding, the gradient of bias is also sharded.
-        if sharded_dim == 0:
-            dist.all_reduce(local_bias_grad)
-            local_bias_grad = local_bias_grad.narrow(
-                sharded_dim, start_pos, chunk_size
-            )
-            sharded_bias_grad = sharded_bias_grad.narrow(
-                sharded_dim, start_pos, chunk_size
-            )
+        dist.all_reduce(local_bias_grad)
```

Comment:
We shouldn't need this based on our discussion of how we can use an autograd function for bias calculation?

Reply:
This is still needed. As we did the same for weight. Local grad needs to be aggregated across different ranks because each rank only has one input.